java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,494 @@
+      if (programPosition.i < programPosition.block.elements().size()) {
+        // process block element
+        visit(programPosition.block.elements().get(programPosition.i), programPosition.block.terminator());
+      } else if (programPosition.block.terminator() == null) {
+        // process block exit, which is unconditional jump such as goto-statement or return-statement
+        handleBlockExit(programPosition);
+      } else if (programPosition.i == programPosition.block.elements().size()) {
+        // process block exist, which is conditional jump such as if-statement
+        checkerDispatcher.executeCheckPostStatement(programPosition.block.terminator());
+      } else {
+        // process branch
+        handleBlockExit(programPosition);
+      }
+    }
+
+    checkerDispatcher.executeCheckEndOfExecution(tree);
+    // Cleanup:
+    explodedGraph = null;
+    workList = null;
+    node = null;
+    programState = null;
+    constraintManager = null;
+  }
+
+  private Iterable<ProgramState> startingStates(MethodTree tree, ProgramState ps) {
+    Iterable<ProgramState> startingStates = Lists.newArrayList(ps);
+    for (final VariableTree variableTree : tree.parameters()) {
+      // create
+      final SymbolicValue sv = constraintManager.createSymbolicValue(variableTree);
+      startingStates = Iterables.transform(startingStates, new Function<ProgramState, ProgramState>() {
+        @Override
+        public ProgramState apply(ProgramState input) {
+          return ProgramState.put(input, variableTree.symbol(), sv);
+        }
+      });
+
+      if (variableTree.symbol().metadata().isAnnotatedWith("javax.annotation.CheckForNull")
+        || variableTree.symbol().metadata().isAnnotatedWith("javax.annotation.Nullable")) {
+        startingStates = Iterables.concat(Iterables.transform(startingStates, new Function<ProgramState, List<ProgramState>>() {
+          @Override
+          public List<ProgramState> apply(ProgramState input) {
+            List<ProgramState> states = new ArrayList<>();
+            states.addAll(sv.setConstraint(input, NullConstraint.NULL));
+            states.addAll(sv.setConstraint(input, NullConstraint.NOT_NULL));
+            return states;
+          }
+        }));
+
+      }
+    }
+    return startingStates;
+  }
+
+  private void handleBlockExit(ExplodedGraph.ProgramPoint programPosition) {
+    CFG.Block block = programPosition.block;
+    Tree terminator = block.terminator();
+    if (terminator != null) {
+      switch (terminator.kind()) {
+        case IF_STATEMENT:
+          handleBranch(block, ((IfStatementTree) terminator).condition());
+          return;
+        case CONDITIONAL_OR:
+        case CONDITIONAL_AND:
+          handleBranch(block, ((BinaryExpressionTree) terminator).leftOperand());
+          return;
+        case CONDITIONAL_EXPRESSION:
+          handleBranch(block, ((ConditionalExpressionTree) terminator).condition());
+          return;
+        case FOR_STATEMENT:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)